addon/mixins/run.js

```diff
@@ -92,6 +92,54 @@ export default Mixin.create({
   },
 
   /**
+   Adds the provided function to the named queue to be executed at the end of the RunLoop.
+   The timer is properly canceled if the object is destroyed before it is invoked.
+
+   Example:
+
+   ```js
+   import Component from 'ember-component';
+   import ContextBoundTasksMixin from 'web-client/mixins/context-bound-tasks';
+
+   export default Component.extend(ContextBoundTasksMixin, {
+     init() {
+       this._super(...arguments);
+       this.scheduleTask('afterRender', () => {
+         console.log('This runs at the end of the run loop (via the afterRender queue) if this component is still displayed');
+       })
+     }
+   });
+   ```
+
+   @method scheduleTask
+   @param { Function } callback the callback to run at the provided time
+   @param { ...* } args arguments to pass to the callback
+   @param { Number } [timeout=0] the time in the future to run the callback
+   @public
+   */
+  scheduleTask(queue, callbackOrName, ...args) {
+    assert(`Called \`scheduleTask\` without a string as the first argument on ${this}.`, typeof queue === 'string');
+    assert(`Called \`scheduleTask\` on destroyed object: ${this}.`, !this.isDestroyed);
+    let type = typeof callbackOrName;
+
+    let cancelId = run.schedule(queue, this, () => {
+      let cancelIndex = this._pendingTimers.indexOf(cancelId);
+      this._pendingTimers.splice(cancelIndex, 1);
```

Comment:
In this case I wonder if a lazy no-op of the scheduled function might be a better performance option than explicit cancellation. Explicit cancellation means walking the list of scheduled tasks to find the cancelled one, then modifying the queue. A lazy no-op would simply mean the scheduled task sits on the queue until it run, then it determines it doesn't need to and truncates.